test/datalog_tests/json_test.dl

```diff
@@ -0,0 +1,107 @@
+import json
+
+output relation JsonTest(description: string, value: string)
+
+
+typedef BoolStruct = BoolStruct {
+    b: bool
+}
+
+function boolStruct1(): string = [|{"b":true}|]
+function boolStruct2(): string = [|{"b":true, "foo":"bar"}|]
+function boolStruct3(): string = [|{"foo":"bar"}|]
+
+JsonTest(boolStruct1(),
+         result_unwrap_or_default(to_json_string(from_json_string(boolStruct1()): Result<BoolStruct, string>))).
```

Comment:
This Look like it could use a little helper function.

Reply:
Unfortunately, a function won't work, as I need to sneak the type annotations in-between the two transformations (otherwise it's just a function from strings to strings and there is no way for the compiler to figure out what the deserialized type should be).

I guess I can at least use

``` function to_json_string_or_default(x: 'T): string```